File: forcetracker333_ui/src/services/CommandCenterService.js
import http from "../http-common";

class CommandCenterService {
  getAllCommandCenters(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/commandCenter/commandCenters`, searchDTO);
  }

  get(commandCenterId) {
    return this.getRequest(`/commandCenter/${commandCenterId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/commandCenter?field=${matchData}`, null);
  }

  addCommandCenter(data) {
    return http.post("/commandCenter/addCommandCenter", data);
  }

  update(data) {
  	return http.post("/commandCenter/updateCommandCenter", data);
  }
  
  uploadImage(data,commandCenterId) {
  	return http.postForm("/commandCenter/uploadImage/"+commandCenterId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new CommandCenterService();

File: src/main/java/com/forcetracker333/service/CommandCenterService.java
package com.forcetracker333.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.forcetracker333.domain.CommandCenter;
import com.forcetracker333.dto.CommandCenterDTO;
import com.forcetracker333.dto.CommandCenterSearchDTO;
import com.forcetracker333.dto.CommandCenterPageDTO;
import com.forcetracker333.dto.CommandCenterConvertCriteriaDTO;
import com.forcetracker333.service.GenericService;
import com.forcetracker333.dto.common.RequestDTO;
import com.forcetracker333.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface CommandCenterService extends GenericService<CommandCenter, Integer> {

	List<CommandCenter> findAll();

	ResultDTO addCommandCenter(CommandCenterDTO commandCenterDTO, RequestDTO requestDTO);

	ResultDTO updateCommandCenter(CommandCenterDTO commandCenterDTO, RequestDTO requestDTO);

    Page<CommandCenter> getAllCommandCenters(Pageable pageable);

    Page<CommandCenter> getAllCommandCenters(Specification<CommandCenter> spec, Pageable pageable);

	ResponseEntity<CommandCenterPageDTO> getCommandCenters(CommandCenterSearchDTO commandCenterSearchDTO);
	
	List<CommandCenterDTO> convertCommandCentersToCommandCenterDTOs(List<CommandCenter> commandCenters, CommandCenterConvertCriteriaDTO convertCriteria);

	CommandCenterDTO getCommandCenterDTOById(Integer commandCenterId);







}





